lib/block.rb

```diff
@@ -0,0 +1,17 @@
+require 'date'
+require_relative 'reservation'
+
+module HotelManagementSystem
+    class Block
+        attr_reader :start_date, :end_date, :room_collection, :discount_rate
+
+        def initialize(start_date:, end_date:, room_collection: [], discount_rate: 0.8)
```

Comment:
I would have liked to see `Block` use a `DateRange` object internally instead of just having a start and end date.